test/e2e/metricscompare_test.go

```diff
@@ -0,0 +1,102 @@
+package e2e
+
+import (
+	"fmt"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// verifyCheckerResultMetricsValueIncreased is a helper function to verify if metrics corresponding to checker results have increased from
+// time0 to timeN. The function is compatible with the cluster_health_monitor_pod_health_result_total and cluster_health_monitor_checker_result_total
+// metrics. For every provided checker name, it will check whether the metric with desired type, status, and error code has increased.
+// Returns true only if there is an increase for every metric checked, false otherwise. It also returns a slice containing every checker name
+// that increased (for logging/debug purposes), and an error if any comparison fails.
+func verifyCheckerResultMetricsValueIncreased(time0Metrics, timeNMetrics map[string]*dto.MetricFamily, metricName string, checkerNames []string, checkerType, status, errorCode string) (bool, []string, error) {
```

Comment:
is the second return always equal to the input parameters `checkerNames` if the first return is true? if that is true, I think we don't need return the second value.